File: CS/WindowsFormsApplication1/Form1.cs
using System;
using System.ComponentModel;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraEditors.Controls;
using DevExpress.Utils;

namespace WindowsFormsApplication1 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }
        private void Form1_Load(object sender, EventArgs e) {
            BindingList<PictureObject> list = new BindingList<PictureObject>();
           list.Add(new PictureObject("Image0", @"https://i.imgur.com/tHyzhwKb.jpg"));
            list.Add(new PictureObject("Image1", @"https://i.imgur.com/tHyzhwKb.jpg"));
            list.Add(new PictureObject("Image2", @"https://i.imgur.com/tHyzhwKb.jpg"));
            gridView1.OptionsView.RowAutoHeight = true;
            gridView1.OptionsView.AnimationType = DevExpress.XtraGrid.Views.Base.GridAnimationType.AnimateAllContent;
            gridControl1.DataSource = list;
            
            list.ListChanged += (s, args) => { 
                if(args.PropertyDescriptor.Name == "Image")
                    gridView1.LayoutChanged(); 
            };
        }
        public class PictureObject : INotifyPropertyChanged {
            public string Name {get; set;}
            public Image Image {get; set;}
            public event PropertyChangedEventHandler PropertyChanged;
            public PictureObject(string name, string url) {
                Name = name;
                Image = ResourceImageHelper.CreateImageFromResources("DevExpress.XtraEditors.Images.loading.gif", typeof(BackgroundImageLoader).Assembly);
                BackgroundImageLoader bg = new BackgroundImageLoader();
                 bg.Loaded += (s, e) => {
                     Image = bg.ImageObject as Image;
                    if (!(Image is Image)) Image = ResourceImageHelper.CreateImageFromResources("DevExpress.XtraEditors.Images.Error.png", typeof(BackgroundImageLoader).Assembly);
                    PropertyChanged(this, new PropertyChangedEventArgs("Image"));
                    //bg.Dispose();
                };
                bg.Load(url);
            }
        }
    }
}

File: VB/WindowsFormsApplication1/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.Controls
Imports DevExpress.Utils

Namespace WindowsFormsApplication1

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            Dim list As BindingList(Of PictureObject) = New BindingList(Of PictureObject)()
            list.Add(New PictureObject("Data Grid", "https://www.devexpress.com/Products/NET/Controls/WinForms/i/landing/controls/WinForms-Data-Grid-VisualStudio2013-Light-Theme.png"))
            list.Add(New PictureObject("Loading Error", "https://www.devexpress.com/notfound.png"))
            list.Add(New PictureObject("Banded Reporting", "https://www.devexpress.com/Products/NET/Controls/WinForms/i/landing/controls/Reporting-VisualStudio2013-Light-Theme.png"))
            gridView1.OptionsView.RowAutoHeight = True
            gridView1.OptionsView.AnimationType = DevExpress.XtraGrid.Views.Base.GridAnimationType.AnimateAllContent
            gridControl1.DataSource = list
            AddHandler list.ListChanged, Sub(s, args)
                If Equals(args.PropertyDescriptor.Name, "Image") Then gridView1.LayoutChanged()
            End Sub
        End Sub

        Public Class PictureObject
            Implements INotifyPropertyChanged

            Public Property Name As String

            Public Property Image As Image

            Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

            Public Sub New(ByVal name As String, ByVal url As String)
                Me.Name = name
                Image = ResourceImageHelperCore.CreateImageFromResources("DevExpress.XtraEditors.Images.loading.gif", GetType(BackgroundImageLoader).Assembly)
                Dim bg As BackgroundImageLoader = New BackgroundImageLoader()
                AddHandler bg.Loaded, Sub(s, e)
                    Image = TryCast(bg.ImageObject, Image)
                    If Not(TypeOf Image Is Image) Then Image = ResourceImageHelperCore.CreateImageFromResources("DevExpress.XtraEditors.Images.Error.png", GetType(BackgroundImageLoader).Assembly)
                    RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs("Image"))
                    bg.Dispose()
                End Sub
                bg.Load(url)
            End Sub
        End Class
    End Class
End Namespace
